x/pairing/keeper/pairing.go

```diff
@@ -144,13 +147,19 @@ func (k Keeper) calculatePairingForClient(ctx sdk.Context, providers []epochstor
 		validProviders = append(validProviders, stakeEntry)
 	}
 
-	// calculates a hash and randomly chooses the providers
-	servicersToPairCount, err := k.ServicersToPairCount(ctx, epochStartBlock)
-	if err != nil {
-		return nil, nil, err
+	spec, found := k.specKeeper.GetSpec(ctx, chainID)
+	if !found {
+		return nil, nil, fmt.Errorf("spec not found or not enabled")
 	}
-	validProviders = k.returnSubsetOfProvidersByStake(ctx, clientAddress, validProviders, servicersToPairCount, epochStartBlock, chainID)
 
+	if spec.ProvidersTypes == spectypes.Spec_dynamic {
+		// calculates a hash and randomly chooses the providers
+		servicersToPairCount, err := k.ServicersToPairCount(ctx, epochStartBlock)
+		if err != nil {
+			return nil, nil, err
+		}
+		validProviders = k.returnSubsetOfProvidersByStake(ctx, clientAddress, validProviders, servicersToPairCount, epochStartBlock, chainID)
+	}
 	for _, stakeEntry := range validProviders {
```

Comment:
IMPORTANT: 
this will cause relay payment CU limit to malfunction, since you didn't change k.ServicersToPairCount(ctx, epochStartBlock) in relay payment CU limits calculations, this is the first time we don't returrn a fixed length of providers

please do another overveiw on all places that use the providers count, check we didn't miss anything

also please check with protocol team if they have code that uses providers length in a way that it might break